Source/Core/InputCommon/ControllerInterface/Device.cpp

```diff
@@ -146,15 +146,43 @@ bool DeviceQualifier::operator==(const DeviceQualifier& devq) const
 
 Device* DeviceContainer::FindDevice(const DeviceQualifier& devq) const
 {
-  for (Device* d : m_devices)
+  std::lock_guard<std::mutex> lk(m_devices_mutex);
+  for (const auto& d : m_devices)
   {
-    if (devq == d)
-      return d;
+    if (devq == d.get())
+      return d.get();
```

Comment:
Can we get rid of the need for `FindDevice` to return a reference to the Device to someone at the outside? If yes, we can stick with a `unique_ptr`. Otherwise it would have to be a `shared_ptr` (or something similar) to avoid reference problems.